### ISSUE TITLE
Unable to access drop down component, while navigating using NVDA browse and Form modes.

### ISSUE THREAD

**Issue by SagarThakurCSE** [microsoft/fluentui#17552]
### Environment Information:

- **Tool**: NVDA
- **Browser version**: New Edge Version 89.0.774.54 (Official build) (64-bit)

### Describe the issue:

Users who rely on screen reader are blocked as they are unable to access drop down, while navigating using NVDA browse and Form modes.

### Please provide a reproduction of the issue in a codepen:

Codepen: https://codepen.io/sathak-MSFT/pen/WNRvqyP

Repro Steps:
Step 1: Enable NVDA and open above codepen in New Edge browser.
Step 2: Navigate to "Basic uncontrolled example 2" drop down using NVDA browse mode and form mode 
Step 3: Verify whether user is able to access "Basic uncontrolled example 2" drop down or not, while navigating using NVDA browse and Form modes.

#### Actual behavior:

Unable to access "Basic uncontrolled example 2" drop down present in the above codepen, while navigating using NVDA browse and Form modes.
Ex: For "Basic uncontrolled example 2" drop down, NVDA is narrating the information in NVDA browse and Form modes as 'Basic uncontrolled example 2 Apple  list  collapsed  subMenu' but unable to access it.
Observation: This control is accessible in focus mode but, when tried switching from browse mode to insert mode (i.e. by selecting 'insert + Space') NVDA Focus is moving to previous focused controls, it is not staying on the same control.

#### Expected behavior:

User should able to access "Basic uncontrolled example 2" while navigating using NVDA browse and Form modes.

### Documentation describing expected behavior

https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html

**Comment by gouttierre** [comment]
@SagarThakurCSE - Thank you for your findings and repo steps.
@TristanWatanabe - Would you be able to confirm if this is a regression, or if this behavior is an issue with the dropdown controller?
cc @smhigley

**Comment by SagarThakurCSE** [comment]
@gouttierre, @TristanWatanabe : Any update on this bug?

If there's no immediate fix, could you please suggest an alternative
